src/web/arr/trove/chart.arr

```diff
@@ -75,13 +86,150 @@ fun to-table3(xs :: List<Any>, ys :: List<Any>, zs :: List<Any>) -> TableIntern:
+       Calculates the resulting Coeff, Exponent where number = coeff * base ^ Exponent.
+       Currently only works with bases > 1.
+       ```
+  when base <= 1: 
+    raise("Num-to-scientific: Only defined on bases > 1")
+  end
+  
+  fun recur(s :: SciNumber): 
+    doc: ``` 
+         Takes the current Coeff, Exponent and divides/multiplies by base to move closer to 
+         the actual scientific representation.
+         ```
+    cases (SciNumber) s: 
+      | sci-notation(c, e, b) => 
+        pos-c = num-abs(c)
+        ask: 
+          | (pos-c > 0) and (pos-c < 1) then: recur(sci-notation(c * b, e - 1, b))
+          | (pos-c == 0) or ((pos-c >= 1) and (pos-c < b)) then: sci-notation(c, e, b)
+          | otherwise: recur(sci-notation(c / b, e + 1, b))
+        end
+    end
+  end
+  
+  {(n): recur(sci-notation(n, 0, base))}
+#|
+where: 
+  num-to-scientific(10)(0) is sci-notation(0, 0, 10)
+  num-to-scientific(10)(3.214) is sci-notation(3.214, 0, 10)
+  num-to-scientific(10)(513) is sci-notation(5.13, 2, 10)
+  num-to-scientific(10)(-23) is sci-notation(-2.3, 1, 10)
+  num-to-scientific(10)(0.00123) is sci-notation(1.23, -3, 10)
+  num-to-scientific(10)(-0.0231) is sci-notation(-2.31, -2, 10)
+  num-to-scientific(2)(256) is sci-notation(1, 8, 2)
+  num-to-scientific(1) raises "Only defined on bases > 1"
+  num-to-scientific(0.32) raises "Only defined on bases > 1"
+  num-to-scientific(0) raises "Only defined on bases > 1"
+  num-to-scientific(-50) raises "Only defined on bases > 1"
+|#
+end
+
+fun prep-axis(values :: List<Number>) -> {Number; Number}: 
+  doc: ``` Calculate the max axis (top) and min axis (bottom) values for bar-chart-series```
+
+  get-with-cmp = {(cmp :: (Number, Number -> Boolean), l :: List<Number>) -> Number: 
+    fold({(acc, elm): 
+      if cmp(acc, elm): acc
+      else: elm
+      end}, l.first, l)}
+
+  max-positive-height = num-max(0, get-with-cmp({(a, b): a > b}, values))
+  max-negative-height = num-min(0, get-with-cmp({(a, b): a < b}, values))
+
+  {max-positive-height; max-negative-height}
+end
+
+fun multi-prep-axis(is-stacked :: String, value-lists :: List<List<Number>>) 
+  -> {Number; Number}: 
+  doc: ``` 
+       Calculate the max axis (top) and min axis (bottom) values for multi-bar-chart-series
+       ```
+
+  get-with-cmp = {(cmp :: (Number, Number -> Boolean), l :: List<Number>) -> Number: 
+    fold({(acc, elm): 
+      if cmp(acc, elm): acc
+      else: elm
+      end}, l.first, l)}
+
+  ask:
+    | is-stacked == 'none' then: 
+      # Find the tallest bar in the entire group 
+      positive-max-groups = map({(l): get-with-cmp({(a, b): a > b}, l)}, value-lists)
+      negative-max-groups = map({(l): get-with-cmp({(a, b): a < b}, l)}, value-lists)
+      max-positive-height = num-max(0, get-with-cmp({(a, b): a > b}, positive-max-groups))
+      max-negative-height = num-min(0, get-with-cmp({(a, b): a < b}, negative-max-groups))
```

Comment:
replace these lambdas with `num-max` and `num-min`, and then eliminate `get-with-cmp` in favor of calling `fold` directly

Reply:
Good point, I guess this was an attempt at generalization that just ended up being unnecessary because we only needed num-min, num-max, and fold